docs/api/proposed_endpoints.md

```diff
@@ -507,3 +528,320 @@ Status: 200 OK
+
+```
+POST /api/swag (requires authentication)
+```
+
+**Request**
+```json
+{
+    "name": "Tshirt",
+    "description": "Sponsered by ABC"
+}
+```
+
+**Response**
+Status: 201 Created
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "Tshirt",
+    "description": "Sponsered by ABC",
+    "image": null,
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+## Update swag details
+
+```
+PATCH /api/swag/:swag_id (requires authentication)
+```
+
+**Request**
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "Pendrive",
+    "description": "Sponsered by XYZ",
+    "image": null,
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+**Response**
+Status: 201 Created
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "Pendrive",
+    "description": "Sponsered by XYZ",
+    "image": null,
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+## Get swag details
+
+```
+GET /api/swag/:swag_id (requires authentication)
+```
+
+**Response**
+Status: 200 OK
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "Tshirt",
+    "description": "Sponsered by ABC",
+    "image": null,
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+## Get details of all swags
+
+```
+GET /api/swag (requires authentication)
+```
+
+**Response**
+Status: 200 OK
+```json
+{
+    "count": 1,
+    "next": null,
+    "previous": null,
+    "results": [
+        {
+            "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+            "name": "Tshirt",
+            "description": "Sponsered by ABC",
+            "image": null,
+            "created_at": "2018-08-01T17:30:42Z",
+            "modified_at": "2018-08-01T17:30:42Z"
+        }
+    ]
+}
+```
+
+## Delete a swag
+
+```
+DELETE /api/swag/:swag_id (requires authentication)
+```
+
+**Response**
+Status: 204 No-Content
+
+## Upload image of a swag
+
+```
+POST /api/swag/:swag_id/upload_image (requires authentication)
+```
+
+Image will be uploaded as multipart data as a streaming HTTP request.
+
+__NOTE__: `image` key will be used to send multipart data.
+
+**Response**
+Status: 201 Created
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "Tshirt",
+    "description": "Sponsered by ABC",
+    "image": "http://xyz.com/url/of/uploaded_image.jpg",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+## Delete the image of swag
+
+```
+DELETE /api/swag/:swag_id/delete_image (requires authentication)
+```
+
+**Response**
+Status: 200 OK
+```json
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "name": "Tshirt",
+    "description": "Sponsered by ABC",
+    "image": null,
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z"
+}
+```
+
+# User-Swag
```

Comment:
What is User-Swag?

Reply:
`UserSwag` is a through-table to relate `Users` with `Swags` as a many-to-many relation. All the following endpoints are dealing on this model. It will exist in the `Swags` app itself and its usage lies in marking if a `Swag` is given to a `User` and the time when it is given.

Reply:
Why are we having an entry in the API for swag when the swag is not given yet?

Reply:
Right! We don't need to create an entry if the swag is not given yet. Following this, we won't be needing `is_given`, because the creation of the entry will itself mean that the swag is given and `created_at` will be the same as `given_at`. So we can remove both `is_given` and `given_at`.

Reply:
👍